File: 03/2.py
def get_code(char):
    o = ord(char)
    if o > 96:
        return o - 96
    return o - 38

with open('input.txt') as f:
    lines = []
    for l in f:
        line = set([c for c in l.replace('\n', '')])
        lines.append(''.join(line))
    
    m = {}
    s = 0
    for i in range(len(lines)):
        if not i % 3:
            m = {}
        l = lines[i]
        for c in l:
            m[c] = 1 if c not in m else m[c] + 1
            if m[c] == 3:
                print(get_code(c), c)
                s += get_code(c)
                break
    print(s)

File: 10/2.mjs
import readline from "node:readline/promises";
import { createReadStream } from "node:fs";

const reader = readline.createInterface({
  input: createReadStream("./input.txt"),
});

const lines = [];

for await (const line of reader) {
  lines.push(line);
}

let X = 1;
let read = true;
let value = 0;

const rows = ["", "", "", "", "", ""];

for (let i = 0; i < 240; i++) {
  const row = Math.floor(i / 40);
  const col = i % 40;

  rows[row] += Math.abs(col - X) > 1 ? " " : "▮";

  if (!read) {
    read = true;
    X += value;
    continue;
  }

  const line = lines.shift();
  if ("noop" === line) {
    continue;
  }

  read = false;
  value = Number(line.split(" ")[1]);
}

// console.log(X);

console.log(rows.join("\n"));
